### PR TITLE
feat: REST interface for node

### PR DESCRIPTION
**Pull request by antmat**
Rest interface for Node.
@3Hren @sshaman1101 @oopcode PTAL

### CODE REVIEW

File: insonmnia/node/server.go [hunk 1]
```diff
@@ -78,20 +80,29 @@ func newRemoteOptions(ctx context.Context, key *ecdsa.PrivateKey, conf Config, c
...
+	ctx     context.Context
+	cancel  context.CancelFunc
+	privKey *ecdsa.PrivateKey
 }
 
 // New creates new Local Node instance
 // also method starts internal gRPC client connections
 // to the external services like Market and Hub
 func New(ctx context.Context, c Config, key *ecdsa.PrivateKey) (*Node, error) {
+	ctx, cancel := context.WithCancel(ctx)
```
Comment (noxiouz) [inline insonmnia/node/server.go]: defer cancel()

File: util/rest/aes.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return &AESDecoderEncoder{
+		cipherBlock: c,
+	}, nil
+}
+
+func (d *AESDecoderEncoder) DecodeBody(request *http.Request) (io.Reader, error) {
+	data, err := ioutil.ReadAll(request.Body)
```
Comment (noxiouz) [inline util/rest/aes.go]: defer request.Close()

Comment (antmat) [reply util/rest/aes.go]: From docs:
	// For server requests the Request Body is always non-nil
	// but will return EOF immediately when no body is present.
	// The Server will close the request body. The ServeHTTP
	// Handler does not need to.

File: insonmnia/node/server.go [hunk 3]
```diff
@@ -78,20 +80,29 @@ func newRemoteOptions(ctx context.Context, key *ecdsa.PrivateKey, conf Config, c
...
+	httpSrv *rest.Server
+	srv     *grpc.Server
+
+	// services, responsible for request handling
+	hub    pb.HubManagementServer
+	market pb.MarketServer
+	deals  pb.DealManagementServer
+	tasks  pb.TaskManagementServer
+
+	//
```
Comment (3Hren) [inline insonmnia/node/server.go]: What?

**Pull request merged**